backend/modules/professor/routers/professor_grading.py

```
@@ -417,3 +418,167 @@ async def revert_to_ai_grade(
+        ).first()
+
+        if not user_progress:
+            raise HTTPException(status_code=404, detail="User progress not found")
+
+        # Get the simulation to check ownership
+        simulation = db.query(Scenario).filter(
+            Scenario.id == user_progress.simulation_id
+        ).first()
+
+        if not simulation:
+            raise HTTPException(status_code=404, detail="Simulation not found")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)